tmt/utils.py

```diff
@@ -6183,3 +6183,29 @@ def __exit__(self, *args: Any) -> None:
     @property
     def duration(self) -> datetime.timedelta:
         return self.end_time - self.start_time
+
+
+def retry(
+        function_to_retry: Callable[..., Any],
+        retry_count: int,
+        delay: int,
+        error_message: str,
+        *args: Optional[Any],
+        **kwargs: Optional[Any]
+        ) -> Any:
```

Comment:
I believe we can also improve the `Any` annotation of the return value:

* as proposed, the `retry` function returns nothing, therefore the annotation can clearly say `None`
* or, `retry` can propagate the return value of `function_to_retry`, which we could write like this:
  ```python
  def retry(function_to_retry: Callable[..., T], ...) -> T:
      ...
      try:
          return function_to_retry(*args, **kwargs)
      ...
  ```

I'd recommend going with the latter - why not propagate the return value to caller? :)